test/integration/acceptance/get_account_test.cpp

```diff
@@ -0,0 +1,247 @@
+ * @when GetAccount is queried on the user
+ * @then there is a valid AccountResponse
+ */
+TEST_F(GetAccount, WithGetDomainPermission) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTxAwait(
+          makeUserWithPerms({interface::permissions::Role::kGetDomainAccounts}),
+          check(1))
+      .sendQuery(makeQuery(), checkValidAccount())
+      .done();
+}
+
+/**
+ * @given a user with only kGetAllAccounts permission
+ * @when GetAccount is queried on the user
+ * @then there is a valid AccountResponse
+ */
+TEST_F(GetAccount, WithGetAllPermission) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTxAwait(
+          makeUserWithPerms({interface::permissions::Role::kGetAllAccounts}),
+          check(1))
+      .sendQuery(makeQuery(), checkValidAccount())
+      .done();
+}
+
+/**
+ * @given a user with all required permissions and a user in the same domain
+ * @when GetAccount is queried on the second user
+ * @then there is a valid AccountResponse
+ */
+TEST_F(GetAccount, WithGetDomainPermissionOtherAccount) {
+  const std::string kUser2Id = kUser2 + "@" + kDomain;
+
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTxAwait(
+          makeUserWithPerms({interface::permissions::Role::kGetDomainAccounts}),
+          check(1))
+      .sendTxAwait(
+          createUserWithPerms(kUser2,
+                              kUser2Keypair.publicKey(),
+                              kRole2,
+                              {interface::permissions::Role::kSetQuorum})
+              .build()
+              .signAndAddSignature(kAdminKeypair)
+              .finish(),
+          check(1))
+      .sendQuery(makeQuery(kUser2Id),
+                 checkValidAccount(kDomain, kUser2Id, kRole2))
+      .done();
+}
+
+/**
+ * @given a user with all required permissions and a user in other domain
+ * @when GetAccount is queried on the second user
+ * @then there is a valid AccountResponse
+ */
+TEST_F(GetAccount, WithGetDomainPermissionOtherAccountInterdomain) {
+  const std::string kNewDomain = "newdom";
+  const std::string kUser2Id = kUser2 + "@" + kNewDomain;
+  auto make_second_user =
+      baseTx()
+          .creatorAccountId(IntegrationTestFramework::kAdminId)
+          .createRole(kRole2, {interface::permissions::Role::kSetQuorum})
+          .createDomain(kNewDomain, kRole2)
+          .createAccount(kUser2, kNewDomain, kUser2Keypair.publicKey())
+          .build()
+          .signAndAddSignature(kAdminKeypair)
+          .finish();
+
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTxAwait(
+          makeUserWithPerms({interface::permissions::Role::kGetAllAccounts}),
+          check(1))
+      .sendTxAwait(make_second_user, check(1))
+      .sendQuery(makeQuery(kUser2Id),
+                 checkValidAccount(kNewDomain, kUser2Id, kRole2))
+      .done();
+}
```

Comment:
[required]
Discussed with @l4l .
Please add missing cases here (cartesian product of three accounts and 4 permission modes) as it was done in #1663. Thanks.